src/server/bitcoin_deposits.rs

```diff
@@ -0,0 +1,77 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::state::database::DatabaseExt;
+use crate::state::AppState;
+use crate::try_start_session;
+use axum::extract::State;
+use axum::http::Request;
+use axum::response::IntoResponse;
+use axum::Json;
+use axum_auto_routes::route;
+use mongodb::bson::doc;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
+
+use super::responses::{ApiResponse, Status};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BitcoinDepositsQuery {
+    starknet_receiving_addresses: Vec<String>,
+}
+
+#[route(get, "/bitcoin_deposits")]
+pub async fn bitcoin_deposits<B>(
+    State(state): State<Arc<AppState>>,
+    request: Request<B>,
+) -> impl IntoResponse {
+    let mut session = try_start_session!(state);
+    if let Err(err) = session.start_transaction().await {
+        return (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            Json(ApiResponse::new(
+                Status::InternalServerError,
+                format!("Database error: {:?}", err),
+            )),
+        );
+    };
+
+    let query_string = request.uri().query().unwrap_or_default();
```

Comment:
can't it be done with serde automatically via BitcoinDepositsQuery? Otherwise you might want to remove the struct

Reply:
When I use it through serde with this kind of query `http://0.0.0.0:8081/bitcoin_to_starknet_mapping?bitcoin_addresses=123&bitcoin_addresses=456` I get this error : `Failed to deserialize query string: invalid type: string "123", expected a sequence`